src/gateway.h

```diff
@@ -71,6 +71,7 @@ struct handle
 	size_t db_id;            /* For use by prepare callback */
 	size_t stmt_id;          /* For use by prepare callback */
 	const char *sql;
+	unsigned exec_count;
```

Comment:
I didn't look to hard, but I find the use of `exec_count` a bit obscure. It seems to somehow replace the previous `gateway->stmt`, which was used to save temporarily prepared statements for `QUERY_SQL` and `EXEC_SQL`, but I don't quite understand `exec_count`.

Reply:
Or maybe I'm confused, I'll try to give another look tomorrow. Don't block on this, I guess it's mostly for my understanding, although some detailed comment in declaration of the `unsigned exec_count` field (and `const char *sql` too) would be great.